src/driver/Firestore/InProcessFirestore.ts

```diff
@@ -283,13 +283,16 @@ export class InProcessFirestoreQuery implements IFirestoreQuery {
         if (_.isNumber(val)) {
             return val
         }
-
+        if (val instanceof InProcessFirestoreDocRef) {
+            return val.id
+        }
```

Comment:
How was this being absent causing an infinite loop?

Reply:
val was being cast to a string which was being returned as `[Object object]` in the following if block
```
if (val && typeof val.toString === "function") {
            val = val.toString()
        }
```

When we passed in a document ref instead of a string value this would occur. When the filtering was happening it was getting "DOCUMENT_ID" and "[Object object]" to compare. 

The code I wrote uses the last document to start after, in this case it kept returning the same document over and over again and would never complete.